### PR TITLE
Move VPA CRDs from Helm chart to a Golang component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind enhancement cleanup
/merge squash

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #2754

**Special notes for your reviewer**:
As part of https://github.com/gardener/gardener/issues/3292#issuecomment-1036058890, the `vpa-tls-certs` secret must be adapted to the secrets manager. I started to use the opportunity and refactor the VPA Helm chart into a Golang component, but given that this change/PR will become quite large, I decided to file this part separately upfront.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/component/vpa/crd.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+func init() {
+	crdResources = map[string]string{
+		"crd-verticalpodautoscalers.yaml":           verticalPodAutoscalerCRD,
+		"crd-verticalpodautoscalercheckpoints.yaml": verticalPodAutoscalerCheckpointCRD,
+	}
+}
+
+type vpaCRD struct {
+	applier  kubernetes.Applier
+	registry *managedresources.Registry
```
Comment (stoyanr) [inline pkg/operation/botanist/component/vpa/crd.go:44]: Why was this `registry` added, what's its purpose? I see that in the only place `NewCRD` is invoked, `registry` is `nil`? Wouldn't it be better to add it when it's actually needed for anything?

Comment (rfranzke) [reply pkg/operation/botanist/component/vpa/crd.go:44]: I expected this question :) It is not yet used as part of this PR, but I will use it later when I open the PR refactoring the VPA Helm chart. The `registry` will be used for managing the VPA CRD when deployed into the shoot. I hope it's fine to keep this commit as part of this PR (the other PR will already be quite large).

**Review by stoyanr — APPROVED** [review]
/lgtm

**Pull request merged**